app/src/features/mocks/modals/ExportMocksModalWrapper/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from "react";
+import { useMocksModalsContext } from "features/mocks/contexts/modals";
+import { ExportMocksModal } from "./ExportMocksModal";
+
+export const ExportMocksModalWrapper: React.FC = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedMockIds, setSelectedMockIds] = useState([]);
+  const [onSuccess, setOnSuccess] = useState(() => () => {});
+
+  const { setOpenShareMocksModalAction } = useMocksModalsContext();
+
+  useEffect(() => {
+    const openModal = (selectedMockIds: string[], onSuccess?: () => void) => {
+      setIsOpen(true);
+      setSelectedMockIds(selectedMockIds);
+      setOnSuccess(() => onSuccess);
+    };
+
+    setOpenShareMocksModalAction(() => openModal);
+  }, [setOpenShareMocksModalAction]);
+
+  const onClose = () => {
+    setIsOpen(false);
+    setSelectedMockIds([]);
+  };
+
+  return isOpen ? (
+    <ExportMocksModal isOpen={isOpen} toggleModal={onClose} selectedMockIds={selectedMockIds} onSuccess={onSuccess} />
```

Comment:
[previous comment](https://github.com/requestly/requestly/pull/2123#discussion_r1740605617) has not been addressed here